tests/integration/modules/test_vmc_vm_disks.py

```diff
@@ -0,0 +1,115 @@
+"""
+    Integration Tests for vmc_vm_disks execution module
+"""
+import pytest
+import requests
+from saltext.vmware.utils import vmc_vcenter_request
+
+
+@pytest.fixture
+def vm_id(salt_call_cli, vmc_vcenter_connect):
+    ret = salt_call_cli.run("vmc_sddc.get_vms", **vmc_vcenter_connect)
+    vm_obj = ret.json[0]
+    return vm_obj["vm"]
+
+
+@pytest.fixture
+def common_data(vmc_vcenter_connect, vm_id):
+    common_data = vmc_vcenter_connect.copy()
+    common_data["vm_id"] = vm_id
+    yield common_data
+
+
+@pytest.fixture
+def request_headers(common_data):
+    return vmc_vcenter_request.get_headers(
+        common_data["hostname"], common_data["username"], common_data["password"]
+    )
+
+
+@pytest.fixture
+def vm_disk_url(common_data):
+    url = "https://{hostname}/api/vcenter/vm/{vm_id}/hardware/disk"
+    return url.format(hostname=common_data["hostname"], vm_id=common_data["vm_id"])
+
+
+@pytest.fixture
+def get_vm_disks(vm_disk_url, common_data, request_headers):
+    session = requests.Session()
+    response = session.get(
+        url=vm_disk_url,
+        verify=common_data["cert"] if common_data["verify_ssl"] else False,
+        headers=request_headers,
+    )
+    response.raise_for_status()
+    return response.json()
```

Comment:
**suggestion** I think that this test will suffer from the same problems that we fixed on the other test. For an integration test it doesn't really make sense to do out-of-band creation, especially since it looks like we're just doing CRUD ops here.

I was chatting about this kind of thing with Caleb on the Salt Core team and with the integration tests, when it's just doing CRUD ops makes sense to do something like this:


```
def test_vm_disks_smoke_test(...):  # maybe there are parameters or fixtures that it should use. Perhaps to generate a random disk name?

    existing_disk = salt_call_cli.run("vmc_vm_disks.get_by_id")
    assert existing_disk is None  # or whatever means it's not there
    
    existing_disks = salt_call_cli.run("vmc_vm_disks.list").json
    assert "some disk id" not in existing_disks  # or something similiar
    
    result = salt_call_cli.run("vmc_vm_disks.create", ...)
    assert result is True  # or whatever
    
    existing_disk = salt_call_cli.run("vmc_vm_disks.get_by_id", expected_id)
    assert existing_disk is not None
    assert existing_disk.id == expected_id
    
    result = salt_call_cli.run("vmc_vm_disks.update", ...)   # some values that you can update
    assert result is True  # or whatever the result should look like
    
    result = salt_call_cli.run("vmc_vm_disks.delete", ... )  # whatever the delete looks like
    
    existing_disk = ...  # Try to get the disk again, now that we've deleted it
    assert existing_disk is None  # or whatever it is that it looks like
```

Basically, since there's not really any logic here - it's just Create, Read, Update, Delete our pattern looks like this:

- Read a non-existent item. Assert that nothing was found
- Create the item brand new. Assert the creation result is correct
- Update the item. Assert the update returned the correct response
- Get the item again. Assert that the correct values have updated.
- Delete the item. Assert that the deletion returns the correct response.
- Get the item again. Assert that it's missing (or whatever the correct state is, for some of them that delete but it doesn't actually delete right away).    

Obviously, this should all change if there's actual logic in the module or state. But if it's basic CRUD? Yeah, that pattern is good :+1:

Reply:
updated tests according to the suggested pattern